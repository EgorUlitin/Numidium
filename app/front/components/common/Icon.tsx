import * as React from 'react'

import { Icon } from 'antd'

import IconEnum from 'util/enum/IconEnum'


interface Props {
    type: IconEnum
    spin?: boolean
    style?: any
}

export default (props: Props) =>
    <Icon type={props.type} spin={props.spin} style={props.style} />
