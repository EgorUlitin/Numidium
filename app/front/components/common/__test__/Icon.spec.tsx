import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Icon, { IconType } from '../Icon'


test('`Icon` renders correctly', () => {
    const tree = renderer
        .create(<Icon type={IconType.ANDROID} />)
        .toJSON()

    expect(tree).toMatchSnapshot()
})
