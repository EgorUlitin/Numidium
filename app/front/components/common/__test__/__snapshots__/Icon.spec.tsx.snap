// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Icon\` renders correctly 1`] = `
<i
  className="anticon anticon-android"
  onClick={undefined}
  style={undefined}
/>
`;
